src/ripple/protocol/STBlob.h

```diff
@@ -156,8 +156,8 @@ class STBlob
     void
     setValue (void const* data, std::size_t size)
     {
-        value_.alloc (size);
-        std::memcpy(value_.data(), data, size);
+        if (value_.alloc (size))
+            std::memcpy(value_.data(), data, size);
```

Comment:
If `data` is null, we have UB even if size is zero.

Reply:
Consider: `if (value_.alloc (size) && data)`

Reply:
If `data` is null, then `size` *should* be 0, and `value_.alloc(size)` will return a false equivalent.
If `data` is null, and `size` is not 0, we have a null dereference, not just UB.
I think `assert(!data == !size);` would be sufficient.

Reply:
So, `vallue_.alloc(size)` will return `nullptr` if `size == 0`. Which means, we'll only invoke `memcpy` if `size != 0` at which point, `data` can't be `nullptr`.

I think that, long term, we should go for `std::fill`/`std::copy`. We could, temporarily, add:

`assert ((size == 0 && data == nullptr) || (size != 0 && data != nullptr))`

or the "fancier"

`assert ((size == 0) == (data == nullptr))`

Reply:
I did not look at the call sites but my comment was based on the potential of the caller passing `data` as null and `size` as greater than zero, which is undefined behavior in `std::memcpy`. The assert is likely all we need.

Reply:
Makes sense. Wilco.